ERCS/erc-7589.md

```diff
@@ -0,0 +1,480 @@
+  assignment.
+* The `isRoleRevocable` function MAY be implemented as pure or view and MUST return whether the grantor can end the role
+  assignment before its expiration date.
+* The `isRoleApprovedForAll` function MAY be implemented as pure or view and MUST return whether the `_operator` is
+  allowed to grant and revoke roles on behalf of the `_grantor`.
+
+> Please note that "approval" refers to allowing users to commit tokens and grant/revoke roles on one's behalf. An
+  approved user either received the role approval or is the target user. Role approvals are not to be confused with
+  [ERC-1155](./eip-1155.md) approvals. More information can be found in the [Role Approvals](#role-approvals) section.
+
+## Rationale
+
+The conventional methods for utilizing SFTs often involve transferring tokens to another address or freezing balances to
+restrict transfers. However, the question arises: How can SFT owners securely delegate the utility of their assets
+without compromising custody?
+
+[ERC-7589](./eip-7589.md) steps in as a solution by introducing the concept of "token commitments". This abstraction
+serves as a powerful tool for users looking to delegate the control of their SFTs. A token commitment can represent
+either a frozen balance or tokens deposited into a contract, offering a standardized and secure way for SFT owners to
+delegate the use of their assets. Through ERC-7589, users gain a versatile mechanism to abstract the complexities of
+secure delegation, enhancing the utility and interoperability of semi-fungible tokens.
+
+[ERC-7589](./eip-7589.md) IS NOT an extension of [ERC-1155](./eip-1155.md). The main reason behind this decision is to
+keep the standard agnostic of any implementation. This approach enables the standard to be implemented externally or on
+the same contract as the SFT and allows dApps to use roles with immutable SFTs.
+
+### Role Approvals
+
+Like [ERC-1155](./eip-1155.md), [ERC-7589](./eip-7589.md) allows users to approve operators to grant and revoke roles on
+their behalf. This feature is crucial for interoperability, as it enables third-party applications to manage user roles
+without custody-level approvals. Role approvals are part of the core interface, and compliant contracts MUST implement
+the `setRoleApprovalForAll` and `isRoleApprovedForAll` functions.
+
+### Automatic Expiration
+
+Automatic expiration is implemented to save users gas. To end a role assignment, instead of requiring users always to
+call `revokeRole`, applications SHOULD call the `roleExpirationDate` and compare it to the current timestamp to check if
+the role is still valid.
+
+In the context of [ERC-7589](./eip-7589.md), dates are represented as `uint64`. The maximum UNIX timestamp represented
+by a `uint64` is about the year 584 billion, which should be enough to be considered "permanent". For this reason, it's
+RECOMMENDED using `type(uint64).max` in use cases where the assignment is required never to expire.
```

Comment:
You should try to keep requirements (items introduced with the ALL CAPS keywords from the RFCs) in the specification section.